File: apps/system/api/domain/workspace/inviter.go
package workspace

import "github.com/ryo034/react-go-template/apps/system/api/domain/workspace/member"

type Inviter struct {
	*member.Member
	workspace *Workspace
}

func NewInviter(m *member.Member, w *Workspace) Inviter {
	return Inviter{m, w}
}

func (i *Inviter) Workspace() *Workspace {
	return i.workspace
}

File: apps/system/client/src/domain/workspace/inviter.ts
import type { Member, Workspace } from "~/domain"
import { Entity } from "~/domain/shared"

interface Props {
  self: Member
  workspace: Workspace
}

export class Inviter extends Entity<Props> {
  static create(v: Props): Inviter {
    return new Inviter(v)
  }

  get self(): Member {
    return this.value.self
  }

  get workspace(): Workspace {
    return this.value.workspace
  }
}
